diff -urN notify-osd-0.9.29.orig//src/defaults.c notify-osd-0.9.29/src/defaults.c
--- notify-osd-0.9.29.orig//src/defaults.c	2010-04-08 16:10:01.000000000 +0800
+++ notify-osd-0.9.29/src/defaults.c	2010-05-02 09:23:26.947902851 +0800
@@ -141,7 +141,7 @@
 #define DEFAULT_PIXELS_PER_EM        10.0f
 #define DEFAULT_SYSTEM_FONT_SIZE     10.0f
 #define DEFAULT_SCREEN_DPI           96.0f
-#define DEFAULT_GRAVITY              GRAVITY_NORTH_EAST
+#define DEFAULT_GRAVITY              GRAVITY_NONE
 
 /* these values are interpreted as milliseconds-measurements and do comply to
  * the visual guide for jaunty-notifications */
